pages/glossary/default-page-language.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Default Page Language
+key: default-page-language
+unambiguous: true
+objective: true
+input_aspects:
+  - DOM tree
+  - Accessibility tree
+  - CSS Styling
+  - Language
+---
+
+The default page language is the _most common language_ used in a [web page][]. If there are no _words_ in the page, the default language is undefined.
+
+The _most common language_ is determined by counting the number of _words_ in the [web page][] that are part of any of the languages in the [language subtag registry][]. The same word can be part of multiple languages. If the number of words is the same, the default language is undefined.
+
+The _words_ used in the following texts should be counted for the _most common language_:
+
+- **page title**: The value of the [document title][] of the [document][] in the [top-level browsing context][].
+- **text nodes**: The value of any [text nodes][] that is [visible][] or [included in the accessibility tree][]
+- **accessible text**: The [accessible name][] and [accessible description][] of any element included in the [accessibility tree][]
+
+**Exception**: Words in text that comes from nodes that have an [ancestor][] in the [flat tree][] with a non-empty (`""`) `lang` attribute, except for the [root node][].
```

Comment:
How is language handled in a case like that:
```html
<div lang="fr"><span id="label">Hello World</span></div>
<img src="foo.jpg" aria-labelledby="label" />
```

With the current definition, "Hello World" is ignored because it has an ancestor with a lang attribute. But how is it handled by AT? Is it going to be read with the ancestor attribute or default to the page language?

Similar question for
```html
<span id="label" lang="fr">Hello World</span>
<img src="foo.jpg" aria-labelledby="label" />
```

Reply:
The text is included because it is part of the accessible name of the img.

Reply:
Then, I do not understand "text that comes from nodes…" the way you seem to (i.e. that need some clarification).

In the examples I've put above, I understand that "Hello World" is text that comes from the `span` node (or more precisely from the text node which is child of the `span`). These nodes do have an ancestor with a non-empty `lang`.

From your answer, I get that you meant that "Hello World" is text that comes from the `img` node, which doesn't have an ancestor with a non-empty `lang`.


Note that the text in the `span` has actually 2 reasons to be counted: it is displayed (and announced), and it is used in an accessible name. One of these reasons is cancelled due to the `lang` attribute, but not the other. That's also a bit counter-intuitive.